platform-sdk/docs/proposals/TSS-Library/TSS-Library.md

```diff
@@ -0,0 +1,812 @@
+
+**Description**: An interface representing the result of a pairing operation, with methods to compare it to other group elements.
+
+**Link**: [PairingResult.java](pairings-api%2FPairingResult.java)
+
+
+#### Swirlds Crypto Pairings Signature Library
+##### Overview
+This module provides cryptography primitives to create EC PublicKeys, EC PrivateKeys, and Signatures.
+
+##### Public API
+###### `SignatureSchema`
+**Link**: [SignatureSchema.java](signature-lib%2FSignatureSchema.java)
+
+**Description**: A pairings signature scheme can be implemented with different types of curves and group assignment configurations.
+For example, two different configurations might consist of a BLS_12_381 curve using G1 of the pairing to generate public key elements or G2 for the same purpose.
+###### `PairingPrivateKey`
+**Link**: [PairingPrivateKey.java](signature-lib%2FPairingPrivateKey.java)
+
+**Description**: A private key generated using the pairings API
+
+###### `PairingPublicKey`
+**Link**: [PairingPublicKey.java](signature-lib%2FPairingPublicKey.java)
+
+**Description**: A public key generated using the pairings API
+###### `PairingSignature`
+**Link**: [PairingSignature.java](signature-lib%2FPairingSignature.java)
+
+**Description**: A signature generated with the private key that can be verified with the public key
+
+###### `GroupAssignment`
+**Link**: [GroupAssignment.java](signature-lib%2FGroupAssignment.java)
+
+**Description**: An enum to clarify which group public keys and signatures are in, for a given SignatureSchema
+
+
+##### Implementation Note
+The serialization of the elements in this module adds a byte to represent the combination of Curve type and group assignment.
+
+##### Examples
+###### Generating a SignatureSchema
+```java
+SignatureSchema signatureSchema = SignatureSchema.from(Curve.ALT_BN128, GroupAssignament.G1_PUBLIC_KEYS);
+```
+###### Generating a Private Key
+```java
+   PairingPrivateKey pk = PairingPrivateKey.create(signatureSchema, new SecureRandom());
+   
+```
+###### Generating a Public Key
+```java
+   PairingPublicKey pbk = PairingPublicKey.create(pk);
+   
+```
+###### Generating a Signature
+```java
+    byte[] message = new byte[]{};
+    PairingSignature s = PairingSignature.sign(pk, message );
+   
+```
+###### Verifying a Signature
+```java
+static{
+    s.verify(pbk, message);
+}
+```
+##### Constraints
+This module will not depend on hedera-services artifacts, so it cannot include logging, metrics, configuration, or any other helper module from that repo.
+##### Dependencies
+swirlds-cryptography-pairings-API and runtime implementation
+##### Other considerations
+Some unknowns are worth investigating in a follow-up task:
+
+#### Swirlds Crypto TSS Library
+##### Overview
+This library implements the Groth21 TSS-specific primitives.
+
+##### Constraints
+Once built, this library will depend on artifacts constructed in other repositories.
+This library would accept Integer.MAX_VALUE -1 participants.
```

Comment:
```suggestion
This library should accept Integer.MAX_VALUE -1 participants.
```

Reply:
`This library accepts Integer.MAX_VALUE -1 participants.`

I think this is an important statement given that it also means we can support Integer.MAX_VALUE -1 nodes

Reply:
Your statement is reasonable, just perhaps a bit in excess of realistic need.

In the real world we'll not likely get anywhere near 2 billion nodes (are there even that many servers in existence right now?) for a very long time.